app/javascript/modules/listingDetailsLottery/ListingDetailsLotterySearchForm.tsx

```diff
@@ -36,7 +36,12 @@ export const ListingDetailsLotterySearchForm = ({
   )
 
   const onSubmit = (data: { lotterySearchNumber: string }) => {
-    const lotterySearchNumber = data.lotterySearchNumber
+    let lotterySearchNumber = data.lotterySearchNumber
+
+    if (lotterySearchNumber.length < 8) {
+      lotterySearchNumber = "0".repeat(8 - lotterySearchNumber.length) + lotterySearchNumber
```

Comment:
[caniuse](https://caniuse.com/mdn-javascript_builtins_string_repeat) is showing only the most recent Chrome on Android supporting `repeat()`, which seems odd.  Also Opera Mobile.  So could do the tried and true `("00000000" + lotterySearchNumber).slice(-8)` to pad it.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/repeat#browser_compatibility

Shows a different result (all compatible) but we should test to ensure